apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationResult.cs

```diff
@@ -32,10 +32,16 @@ public sealed class TranslationResult
         public string TranslatedText { get; }
 
         /// <summary>
-        /// The source language code, either as provided by the caller
-        /// or detected by the server.
+        /// The source language code detected by the server. This may
+        /// be null, typically due to the source language being supplied
+        /// by the caller.
         /// </summary>
-        public string SourceLanguage { get; }
+        public string DetectedSourceLanguage { get; }
+
+        /// <summary>
+        /// The source language code supplied by the caller, if any.
+        /// </summary>
+        public string SpecifiedSourceLanguage { get; }
```

Comment:
Definite ack to the tension here. I suspect having "code" in the documentation (and the fact that the type is `string` rather than `Language`) is enough.